integrations/drift/drift.go

```diff
@@ -0,0 +1,115 @@
+package drift
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/jipiboily/forwardlytics/integrations"
+)
+
+// Drift integration
+type Drift struct {
+	api service
+}
+
+type service interface {
+	request(string, string, []byte) error
+}
+
+type driftAPIProduction struct {
+	baseUrl string
+}
+
+type apiSubscriber struct {
+	Attributes map[string]interface{} `json:"attributes"`
+	CreatedAt  int64                  `json:"createdAt"`
+	UserId     string                 `json:"userId"`
+	OrgId      string                 `json:"orgId"`
+}
+
+type apiEvent struct {
+	OrgId      string                 `json:"orgId"`
+	UserId     string                 `json:"userId"`
+	Event      string                 `json:"event"`
+	CreatedAt  int64                  `json:"createdAt"`
+	Attributes map[string]interface{} `json:"attributes"`
+}
+
+// Identify forwards and identify call to Drift
+func (d Drift) Identify(identification integrations.Identification) (err error) {
+	s := apiSubscriber{}
+	s.UserId = string(identification.UserID)
+	s.CreatedAt = identification.Timestamp
+	s.OrgId = orgID()
+	// Add custom attributes
+	s.Attributes = identification.UserTraits
+	s.Attributes["forwardlyticsReceivedAt"] = identification.ReceivedAt
+	s.Attributes["forwardlyticsTimestamp"] = identification.Timestamp
+	payload, err := json.Marshal(s)
+
+	err = d.api.request("POST", "identify", payload)
+	return
+}
+
+// Track forwards the event to Drift
+func (d Drift) Track(event integrations.Event) (err error) {
+	e := apiEvent{}
+	e.OrgId = orgID()
+	e.UserId = event.UserID
+	e.Attributes = event.Properties
+	event.Properties["forwardlyticsReceivedAt"] = event.ReceivedAt
+	e.Event = event.Name
+	e.CreatedAt = event.Timestamp
+	payload, err := json.Marshal(e)
+	if err != nil {
+		logrus.WithField("err", err).Fatal("Error marshalling drift event to json")
```

Comment:
Use the logging I just mentioned here too. It's not your fault, I did not do a good job originally and when I faced a bug with the Drip integration (not your fault, I was not sending an email for a specific event), I had to add more context.